NMS-14619: Quick Start: Determine service availability (#5350)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/service-assurance/critical-service.adoc

[[ga-service-assurance-critical-service]]
= Improve Monitoring Efficiency

> Monitoring services on an IP network can be resource expensive, especially in cases where many of these services are not available.

Reviewer: ```suggestion
Monitoring services on an IP network can be resource intensive, especially in cases where many of these services are not available.
```

When a service is offline or unreachable, the monitoring system spends most of its time waiting for retries and timeouts.

To improve efficiency, {page-component-title} deems all services on a interface to be down if the critical service is down.
{page-component-title} uses ICMP as the critical service by default.

> The following image shows how to use critical services to generate these events.

Reviewer: ```suggestion
The following image shows how to use critical services to generate these events:
```

> .Service assurance with Pollerd in {page-component-title}

Reviewer 2: ```suggestion
.Service assurance with pollerd in {page-component-title}
```

> .Service assurance with Pollerd in {page-component-title}

Reviewer: ```suggestion
.Service assurance with pollerd in {page-component-title}
```

image::service-assurance/03_node-outage-correlation.png[]

* (1) Critical services are all up on the node and just a `nodeLostService` is sent.
* (2) Critical service of one of many IP interfaces is down and `interfaceDown` is sent.

>       All other services are not tested and no events are sent, the services are assumed to be unreachable.

Reviewer: ```suggestion
All other services are not tested and no events are sent, the services are assumed to be unreachable.
```

* (3) All critical services on the node are down and just a `nodeDown` is sent.

>       All other services on the other IP interfaces are not tested and no events are sent, these services are assumed as unreachable.

Reviewer: ```suggestion
All other services on the other IP interfaces are not tested and no events are sent, these services are assumed as unreachable.
```

{page-component-title} uses the critical service to correlate outages from services to a `nodeDown` or `interfaceDown` event.

> This is a global configuration option of Pollerd, defined in `poller-configuration.xml`.

Reviewer: ```suggestion
This is a global configuration option of pollerd, defined in `poller-configuration.xml`.
```

The {page-component-title} default configuration enables this behavior.

> .Critical service configuration in Pollerd

Reviewer 2: ```suggestion
.Critical service configuration in pollerd
```

> .Critical service configuration in Pollerd

Reviewer: ```suggestion
.Critical service configuration in pollerd
```

      <critical-service name="ICMP" /> <3>
</node-outage>
----
<1> Enable node outage correlation based on a critical service.

> <2> Optional: With nodes without a critical service, this option controls the behavior.

Reviewer 2: ```suggestion
<2> Optional: For nodes without a critical service, this option controls the polling behavior during a node outage.
```

>     If set to `true` then all services will be polled.
>     If set to `false` then the first service in the package that exists on the node will be polled until service is restored, and then polling will resume for all services.

Reviewer: ```suggestion
If set to `true` then all services will be polled.
If set to `false` then the first service in the package that exists on the node will be polled until service is restored, and then polling will resume for all services.
```
